### ISSUE TITLE
mesa-quilt is getting in the way

### ISSUE THREAD

**Issue by uli42** [ArcticaProject/nx-libs#490]
mesa-quilt is driving me nuts. Every time when I want to switch branches I cannot because the patched files get in the way:
```
Please commit your changes or stash them before you switch branches.
```
If I stash them away, switch branches and then switch back I need to run `git stash apply`. This results in timestamp changes and I end up with having to recompile complete mesa even if I am working elsewhere. If I am _not_ stashing them back the code is not compilable anymore.

Also `git add -p` (showing the patched mesa files) and `git rebase -i` (`Cannot rebase: You have unstaged changes.`) cannot be used smoothly after a compile run anymore.

It is very demotivating and feels like wasting my time. Working on several PRs in parallel as I am doing currently is close to impossible in a comfortable way.

Can't we find a smoother solution? 

I think we should at least try not to patch files that git is tracking. Maybe at a makefile rule that copies and patches them and then use the patched ones at all other places.

Update: maybe we can use guilt instead: https://linux.die.net/man/7/guilt

**Comment by sunweaver** [comment]
Won't fix.

The mesa-quilt script is easy to handle... 

```
cd <nxsrcbasedir>
./mesa-quilt pop -a
```

If you build the Debian package, then it is these steps:

```
cd <nxsrcbasedir>
QUILT_PATCHES=debian/patches quilt pop -a
./mesa-quilt pop -a
```
No need to change this for now.
